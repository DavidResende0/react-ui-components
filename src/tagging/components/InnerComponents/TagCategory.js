import React from 'react';
import PropTypes from 'prop-types';
import { OverlayTrigger, Tooltip } from 'patternfly-react';
import Tag from './Tag';
import TaggingPropTypes from '../TaggingPropTypes';

class TagCategory extends React.Component {
  generateTag = tagValue => (
    <Tag
      key={tagValue.id}
      tagCategory={this.props.tagCategory}
      tagValue={tagValue}
      onTagDeleteClick={this.props.onTagDeleteClick}
      truncate={this.props.valueTruncate}
    />
  );

  render() {
    const values = [...this.props.values];
    const categoryTooltip = (
      <Tooltip id="tooltip">{this.props.tagCategory.description}</Tooltip>
    );
    return (
      <ul className="tag-category list-inline">
        <li key={this.props.tagCategory.id}>
          <OverlayTrigger placement="bottom" overlay={categoryTooltip}>
            <div className="category-label">
              {this.props.categoryTruncate(this.props.tagCategory.description)}
            </div>
          </OverlayTrigger>
        </li>
        {values
          .sort((a, b) => (a.description < b.description ? -1 : 1))
          .map(tagValue => this.generateTag(tagValue))}
      </ul>
    );
  }
}

TagCategory.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagCategory: TaggingPropTypes.category,
  values: PropTypes.arrayOf(TaggingPropTypes.category).isRequired,
  categoryTruncate: PropTypes.func,
  valueTruncate: PropTypes.func,
};

TagCategory.defaultProps = {
  categoryTruncate: str =>
    (str.length > 18 ? `${str.substring(0, 18)}...` : str),
  valueTruncate: str => (str.length > 18 ? `${str.substring(0, 18)}...` : str),
};

export default TagCategory;
