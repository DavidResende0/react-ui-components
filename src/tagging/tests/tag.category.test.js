import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import TagCategory from '../components/tagCategory';

const tagCategory = { description: 'animal', id: 1 };
const tagValues = [{ description: 'duck', id: 1 }, { description: 'lion', id: 2 }];
function onDelete(x) {
  return x;
}

describe('TagCategory Component', () => {
  it('match snapshot', () => {
    const tree = mount(<TagCategory tagCategory={tagCategory} tagValues={tagValues} onTagDeleteClick={onDelete} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
