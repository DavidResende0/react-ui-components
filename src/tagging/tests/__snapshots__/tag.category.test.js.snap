// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCategory Component match snapshot 1`] = `
<TagCategory
  onTagDeleteClick={[Function]}
  tagCategory={
    Object {
      "description": "animal",
      "id": 1,
    }
  }
  tagValues={
    Array [
      Object {
        "description": "duck",
        "id": 1,
      },
      Object {
        "description": "lion",
        "id": 2,
      },
    ]
  }
>
  <ul
    className="tag-category list-inline"
  >
    <div
      className="category-label"
    >
      animal
    </div>
    <Tag
      key="1"
      onTagDeleteClick={[Function]}
      tagCategory={
        Object {
          "description": "animal",
          "id": 1,
        }
      }
      tagValue={
        Object {
          "description": "duck",
          "id": 1,
        }
      }
    >
      <li
        className="tag"
        key="1"
      >
        <Label
          bsStyle="primary"
          className="tagColor"
          key="1"
          onRemoveClick={[Function]}
          type="default"
        >
          <Label
            bsClass="label"
            bsStyle="primary"
            className="tagColor"
          >
            <span
              className="tagColor label label-primary"
            >
              duck
              <RemoveButton
                className=""
                onRemoveClick={[Function]}
                title="Remove"
              >
                <a
                  className="pf-remove-button"
                  href="#"
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="pficon pficon-close"
                  />
                  <span
                    className="sr-only"
                  >
                    Remove
                  </span>
                </a>
              </RemoveButton>
            </span>
          </Label>
        </Label>
      </li>
    </Tag>
    <Tag
      key="2"
      onTagDeleteClick={[Function]}
      tagCategory={
        Object {
          "description": "animal",
          "id": 1,
        }
      }
      tagValue={
        Object {
          "description": "lion",
          "id": 2,
        }
      }
    >
      <li
        className="tag"
        key="2"
      >
        <Label
          bsStyle="primary"
          className="tagColor"
          key="2"
          onRemoveClick={[Function]}
          type="default"
        >
          <Label
            bsClass="label"
            bsStyle="primary"
            className="tagColor"
          >
            <span
              className="tagColor label label-primary"
            >
              lion
              <RemoveButton
                className=""
                onRemoveClick={[Function]}
                title="Remove"
              >
                <a
                  className="pf-remove-button"
                  href="#"
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="pficon pficon-close"
                  />
                  <span
                    className="sr-only"
                  >
                    Remove
                  </span>
                </a>
              </RemoveButton>
            </span>
          </Label>
        </Label>
      </li>
    </Tag>
  </ul>
</TagCategory>
`;
